_posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc

```diff
@@ -0,0 +1,723 @@
+form a defense-in-depth set of best practices that mitigate the most common attacks against systems and networks."
+Organizations seeking to implement a comprehensive security program or framework can use the controls to measure their
+progress and prioritize security efforts. The Foundations Benchmark is just one of several guidelines that can help
+reach the bar set by the CIS Controls. Refer to the Benchmark document directly to view how the recommendations map to
+controls.
+
+=== Recommendation sections
+
+==== Identity and Access Management
+_Number of recommendations: 22_
+
+The recommendations in this section involve the use of identity, accounts, authentication, and authorization.
+On AWS, most identity and access control related concerns are managed using the
+link:https://aws.amazon.com/iam/[eponymous IAM service]. Hence, most (but not all) of the recommendations in
+this section discuss particular IAM configurations, such as the configuration of the password policy, the use
+of various groups and roles, and the configuration of multi-factor authentication (MFA) devices.
+
+==== Logging
+_Number of recommendations: 9_
+
+AWS has a variety of logging, monitoring, and auditing features, and the Benchmark has recommendations for
+several of them:
+
+* link:https://aws.amazon.com/cloudtrail/[AWS CloudTrail] tracks user activity and API usage
+* link:https://aws.amazon.com/config/[AWS Config] records and evaluates resource configurations
+* link:https://docs.aws.amazon.com/vpc/latest/userguide/flow-logs.html[VPC Flow Logs] capture network traffic information
+  in VPCs
+
+AWS has several other logging related features that are not covered directly by the Benchmark. For example,
+the primary log ingestion and query service, link:https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/WhatIsCloudWatchLogs.html[Amazon CloudWatch
+Logs], is integrated with many other AWS services. The Benchmark does not contain any recommendations
+specifically for CloudWatch Logs, though many recommendations do make mention of it.
+
+==== Monitoring
+_Number of recommendations: 14_
+
+Monitoring is an overloaded term in the industry. In the context of the AWS Foundations Benchmark, the
+monitoring section is exclusively about monitoring for specific API calls using the CloudTrail service paired
+with link:https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/MonitoringLogData.html[CloudWatch Logs
+filter metrics]. Each recommendation in this section spells out a specific filter and an associated alarm.
+
+==== Networking
+_Number of recommendations: 4_
+
+The Benchmark is uncomfortably light on networking, considering its central role in the security of any
+distributed system. The recommendations merely limit traffic from the zero network (`0.0.0.0/0`) and
+restrict routing for VPC peering connections.
+
+[[production_grade_design]]
+== Production-grade design
```

Comment:
As mentioned above, this section ideally describes the production-grade design in a way anyone, customer or not, can understand and use... The IaC Lib specific details are saved until the "deployment walkthrough", which actually makes use of the code in the IaC Lib.

For example, this section could go through all the benchmark requirements and talk about how we implement them: e.g., implement this one by configuring AWS Config using these settings; implement this other thing by enabling Flow Logs; etc.

Reply:
I rewrote the production grade design section to be more suitable for non-customers. I moved the existing material to a new section in the walkthrough entitled "The Gruntwork Solution". This was a major update compared to the previous version of the guide - hopefully folks find value in this content.